src/conv/k-g/k-g.cpp

```diff
@@ -82,45 +130,109 @@ int main
 			    }
 
 			    double averageThick = (thick1 + thick2 + thick3 + thick4) / 4.0;
-			    bot.setThickness(averageThick * factor);
+			    geometry.setThickness(averageThick * factor);
 			}
 			else
 			    std::cout << "Missing section to part" << partName.c_str() << '\n';
 
 			for (std::set<int>::iterator itr = (it->second).elements.begin(); itr != (it->second).elements.end(); itr++) {
-			    point_t point1;
-			    point_t point2;
-			    point_t point3;
-			    point_t point4;
-
-			    int     n1 = kData.elements[*itr].node1;
-			    int     n2 = kData.elements[*itr].node2;
-			    int     n3 = kData.elements[*itr].node3;
-			    int     n4 = kData.elements[*itr].node4;
-
-			    point1[X] = kData.nodes[n1].x * factor;
-			    point1[Y] = kData.nodes[n1].y * factor;
-			    point1[Z] = kData.nodes[n1].z * factor;
-
-			    point2[X] = kData.nodes[n2].x * factor;
-			    point2[Y] = kData.nodes[n2].y * factor;
-			    point2[Z] = kData.nodes[n2].z * factor;
-
-			    point3[X] = kData.nodes[n3].x * factor;
-			    point3[Y] = kData.nodes[n3].y * factor;
-			    point3[Z] = kData.nodes[n3].z * factor;
-
-			    point4[X] = kData.nodes[n4].x * factor;
-			    point4[Y] = kData.nodes[n4].y * factor;
-			    point4[Z] = kData.nodes[n4].z * factor;
-
-			    if (n3==n4) {
-				bot.addTriangle(point1, point2, point3);
+			    if (kData.elements[*itr].nodes.size() == 4) {
+
+				point_t point1;
+				point_t point2;
+				point_t point3;
+				point_t point4;
+
+				int     n1 = kData.elements[*itr].nodes[0];
+				int     n2 = kData.elements[*itr].nodes[1];
+				int     n3 = kData.elements[*itr].nodes[2];
+				int     n4 = kData.elements[*itr].nodes[3];
+
+				point1[X] = kData.nodes[n1].x * factor;
+				point1[Y] = kData.nodes[n1].y * factor;
+				point1[Z] = kData.nodes[n1].z * factor;
+
+				point2[X] = kData.nodes[n2].x * factor;
+				point2[Y] = kData.nodes[n2].y * factor;
+				point2[Z] = kData.nodes[n2].z * factor;
+
+				point3[X] = kData.nodes[n3].x * factor;
+				point3[Y] = kData.nodes[n3].y * factor;
+				point3[Z] = kData.nodes[n3].z * factor;
+
+				point4[X] = kData.nodes[n4].x * factor;
+				point4[Y] = kData.nodes[n4].y * factor;
+				point4[Z] = kData.nodes[n4].z * factor;
+
+				if (n3 == n4) {
+				    geometry.addTriangle(point1, point2, point3);
+				}
+				else {
+				    geometry.addTriangle(point1, point2, point3);
+				    geometry.addTriangle(point1, point3, point4);
+				}
 			    }
-			    else {
-				bot.addTriangle(point1, point2, point3);
-				bot.addTriangle(point1, point3, point4);

```

Comment:
Is here an issue with a closing bracket?

Reply:
No, I don't think so, I have tested the code with a solid element and shell element. I will retest now to see if there is something wrong

Reply:
I just tested it, no problems.